PR: Port CIs to GitHub actions, deploy to existing repo and bump theme and pre-commit hooks

CAM-Gerlach opened the pull request:
Counterpart to spyder-ide/lektor-icon#17

Ports the services previously on Travis CI (build, check and deploy) to Github Actions, following the same approach as spyder-ide/lektor-icon#17 , since Travis changed their ToS to no longer be free for open source projects. This also has the following benefits:

* Build, check and deploy are now separate, parallel jobs that run much faster
* The common steps of each action are now modular, easily reusable components
* The check jobs no longer run on push (so e.g. a transient check failure doesn't break a deploy, as has happened in the past), while the deploy job is skipped on PRs
* The jobs each indicate their status separately right on Github, so its clear if the failure occurred in the build, check or deploy
* These improvements plus Github's much greater speed and no queue times should result in much quicker CI completion overall

Made as a branch of the upstream repo to trigger Github actions to actually run on the PR.

However, I won't merge this until you give the go-ahead, @ccordoba12 , because this relies on one simple but non-trivial change: eliminating the now essentially redundant separate `develop` and `production` repos and just deploying the site to the `gh-pages` branch here. This makes the need to recreate and maintain the keys and CI access to those separate repos a non-issue, greatly simplifies the deploy configuration, ensures our scripts and actions are consistent between repos, halves the CI time and deploy latency, and significantly reduces the overall maintenance burden. Having separate branches and deploy repos made sense before we had Netlify or could even build the site properly locally, and the deploy process was more complex and relied on less proven tooling, but now that we have all that, there isn't really a clear need for the extra complexity anymore.

Therefore, I propose that, on your approval, I execute the following sequence of actions:

* [x] Create an empty `gh-pages` branch and enable Github pages
* [x] Eliminate `production` and rename `develop` -> `master`
* [x] Update the PR target, branch protection, etc accordingly
* [ ] Merge this PR to `master`
* [ ] Once we confirm the site deploys and is working on the GH-pages domain, you cut over the DNS records for our `spyder-ide.org` domain to point to this repo's Github pages, and delete the `develop.spyder-ide.org` domain

ccordoba12 commented:
Hey @CAM-Gerlach, thanks for working on this. I agree with your proposed changes, so please go ahead and merge them when they are ready.

CAM-Gerlach commented:
I'm not sure why Netlify isn't reporting its status; the only change since the last build, which worked, is a few tweaks to the Readme, but since it passed on that and we need to merge to test the final step, I'll just go ahead now.

CAM-Gerlach commented:
All done, its deployed and live now. @ccordoba12, please remove the DNS `CNAME` record for `develop.spyder-ide.org` as its no longer used and a minor vulnerability to leave open. Thanks.
